File: src/test/java/com/github/sarhatabaot/kraken/core/db/JooqUtilTest.java
package com.github.sarhatabaot.kraken.core.db;

import org.jooq.SQLDialect;
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.*;

/**
 * @author sarhatabaot
 */
class JooqUtilTest {
    
    @Test
    void getDialect() {
        assertEquals(SQLDialect.DEFAULT, JooqUtil.getDialect("defaultTest"));
        assertEquals(SQLDialect.MYSQL, JooqUtil.getDialect("mysql"));
        assertEquals(SQLDialect.MYSQL, JooqUtil.getDialect("mYSql"));
        
        assertEquals(SQLDialect.SQLITE,JooqUtil.getDialect("sqlite"));
        assertEquals(SQLDialect.MARIADB,JooqUtil.getDialect("mariadb"));
    }
}

File: src/test/kotlin/com/github/sarhatabaot/kraken/core/db/JooqUtilTest.kt
package com.github.sarhatabaot.kraken.core.db

import com.github.sarhatabaot.kraken.core.db.JooqUtil.getDialect
import org.jooq.SQLDialect
import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Test

/**
 *
 * @author sarhatabaot
 */
class JooqUtilTest {

    @Test
    fun getDialect() {
        assertEquals(SQLDialect.DEFAULT, getDialect("defaultTest"))
        assertEquals(SQLDialect.MYSQL, getDialect("mysql"))
        assertEquals(SQLDialect.MYSQL, getDialect("mYSql"))

        assertEquals(SQLDialect.SQLITE, getDialect("sqlite"))
        assertEquals(SQLDialect.MARIADB, getDialect("mariadb"))
    }
}